Added new post resilience4j-grpc

alicanhaman opened the pull request:
Wrote a small blog post detailing resilience4j integration with grpc client. An issue I faced when working on the onboarding project. 
Please take a look and comment :smile:

onrylmz commented on _posts/2020-12-17-Resilience4j-grpc.md:
```diff
@@ -0,0 +1,187 @@
+      <groupId>net.devh</groupId>
+      <artifactId>grpc-client-spring-boot-starter</artifactId>
+  </dependency>
+  <!-- basically your grpc proto project -->
```
I'd give more detail regarding proto and grpc relation

alicanhaman replied:
Wrote some content on that and gave reference to baeldung guide.

onrylmz commented on _posts/2020-12-17-Resilience4j-grpc.md:
```diff
@@ -0,0 +1,187 @@
+
+    @GrpcClient("simple-service")
+    private SimpleServiceGrpc.SimpleServiceBlockingStub simpleServiceBlockingStub;
+    private static final String simple_SERVICE = "simpleService";
```
SIMPLE_SERVICE??

fatihafizoglu commented on _posts/2020-12-17-Resilience4j-grpc.md:
```diff
@@ -0,0 +1,187 @@
+
+First lets briefly mention our stack: 
+
+- **Resilience4j** is a lightweight fault tolerance library inspired by [Netflix Hystrix](https://github.com/Netflix/Hystrix), but designed for Java 8 and functional programming. Lightweight, because the library only uses Vavr, which does not have any other external library dependencies. Netflix Hystrix, in contrast, has a compile dependency to Archaius which has many more external library depend ...
```
Give reference to: https://github.com/resilience4j/resilience4j#1-introduction

fatihafizoglu commented on _posts/2020-12-17-Resilience4j-grpc.md:
```diff
@@ -0,0 +1,187 @@
+First lets briefly mention our stack: 
+
+- **Resilience4j** is a lightweight fault tolerance library inspired by [Netflix Hystrix](https://github.com/Netflix/Hystrix), but designed for Java 8 and functional programming. Lightweight, because the library only uses Vavr, which does not have any other external library dependencies. Netflix Hystrix, in contrast, has a compile dependency to Archaius which has many more external library depend ...
+- **gRPC** is a modern open source high performance RPC framework that can run in any environment. It can efficiently connect services in and across data centers with pluggable support for load balancing, tracing, health checking and authentication.
```
Give reference to: https://grpc.io/about/

ertugrulaypek-zz commented on _posts/2020-12-17-Resilience4j-grpc.md:
```diff
@@ -0,0 +1,187 @@
+  </dependency>
+  <dependency>
+      <groupId>io.github.resilience4j</groupId>
+      <artifactId>resilience4j-spring-boot2</artifactId>
```
I think specifying the version of the library would be nice as there might be support for gRPC from resilience4j in the future

alicanhaman replied:
good idea, done.

alicanhaman commented:
Changed the whole SimpleService naming structure to HelloService while writing proto file I notices that it was becoming too confusing. HelloService is the most popular example used in the web and I think this will be easier to understand with that. Also removed converters from service implementation and extra classes to make this easier to just plug in and get working.

v1r7u reviewed (approved):
In general looks good. I see nothing, that could prevent you from posting it :)

Optional recommendations:
- consider adding a bit of explanation about resilience/fault-tolerance and/or circuit breakers. If you do not want to explain it in the article itself, you might mention it in further-reading sections. For example, https://www.usenix.org/conference/srecon18asia/presentation/goh or https://buttondown.email/computer-napkins/archive/napkin-problem-11-circuit-breakers/
- I recommend avoiding using `easy`, `simply`, and others, as what was easy for you, might be quite difficult for someone else. I've seen the same recommendation in doc-guidelines, for example, [Digital Ocean guidelines](https://www.digitalocean.com/community/tutorials/digitalocean-s-technical-writing-guidelines).
- Personally, I prefer when an article talks to the reader. Thus, `In this article, I will explain how to` -> `The article explains`.
- In Final Words, I think, you could just highlight contact information, like `Don't hesitate to contact me via LinkedIn, twitter or email`, but skip the first two sentences.
